File: Graph/java/GraphUtils.java
import java.util.ArrayList;
import java.util.HashMap;
import java.util.List;
import java.util.Map;

public final class GraphUtils {
  private GraphUtils() {
  }

  public static Map<String, List<String>> buildUndirectedGraph(List<String[]> edges) {
    Map<String, List<String>> graph = new HashMap<>();
    for (String[] edge : edges) {
      String a = edge[0];
      String b = edge[1];
      graph.computeIfAbsent(a, key -> new ArrayList<>()).add(b);
      graph.computeIfAbsent(b, key -> new ArrayList<>()).add(a);
    }
    return graph;
  }

  public static boolean inBounds(char[][] grid, int row, int col) {
    return row >= 0 && row < grid.length && col >= 0 && col < grid[0].length;
  }

  public static final int[][] DIRECTIONS = new int[][] {
      {-1, 0}, {1, 0}, {0, -1}, {0, 1}
  };
}

File: Graph/csharp/GraphUtils.cs
using System.Collections.Generic;

namespace GraphSolutions;

public static class GraphUtils
{
    public static Dictionary<string, List<string>> BuildUndirectedGraph(IEnumerable<(string, string)> edges)
    {
        var graph = new Dictionary<string, List<string>>();
        foreach (var (a, b) in edges)
        {
            if (!graph.ContainsKey(a))
            {
                graph[a] = new List<string>();
            }
            if (!graph.ContainsKey(b))
            {
                graph[b] = new List<string>();
            }
            graph[a].Add(b);
            graph[b].Add(a);
        }

        return graph;
    }

    public static bool InBounds(char[][] grid, int row, int col)
    {
        return row >= 0 && row < grid.Length && col >= 0 && col < grid[0].Length;
    }

    public static readonly (int, int)[] Directions =
    {
        (-1, 0),
        (1, 0),
        (0, -1),
        (0, 1),
    };
}
